torch/csrc/api/include/torch/data/transforms/stack.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <torch/data/example.h>
+#include <torch/data/transforms/collate.h>
+#include <torch/tensor.h>
+
+#include <utility>
+#include <vector>
+
+namespace torch {
+namespace data {
+namespace transforms {
+
+template <typename T = Example<>>
+struct Stack;
+
+/// A `Collation` for `Example<Tensor, Tensor>` types that stacks all data
+/// tensors into one tensor, and all target (label) tensors into one tensor.
+template <>
+struct Stack<Example<>> : public Collation<Example<>> {
```

Comment:
Can we plz also stack vectors of T where T can be e.g. an `int`?

Reply:
So `vector<int>` should be stacked to `vector<vector<int>>`?

Reply:
If it were to follow the python interface, it would stack a `vector<int>` into a `Tensor` of `int64_t`.